Adding a Connectivity.ConnectivityChanged handler throw an exception on Android 14

### Description

When adding a Connectivity.ConnectivityChanged handler, the following exception is thrown on an Android 14 device, when the targetSdkVersion is 34 (the default with new Maui app) an when run on Android 14:

```
Exception has occurred: Java.Lang.SecurityException
com.companyname.dotnetmaui: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts
    at Java.Interop.JniEnvironment.InstanceMethods.CallNonvirtualObjectMethod(JniObjectReference instance, JniObjectReference type, JniMethodInfo method, JniArgumentValue* args) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/obj/Release/net7.0/JniEnvironment.g.cs:20416
    at Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeVirtualObjectMethod(String encodedMember, IJavaPeerable self, JniArgumentValue* parameters) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.JniInstanceMethods_Invoke.cs:876
    at Android.Content.ContextWrapper.RegisterReceiver(BroadcastReceiver receiver, IntentFilter filter) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Android.Content.ContextWrapper.cs:2605
    at Microsoft.Maui.Networking.ConnectivityImplementation.StartListeners() in D:\a\_work\1\s\src\Essentials\src\Connectivity\Connectivity.android.cs:48
    at Microsoft.Maui.Networking.ConnectivityImplementation.add_ConnectivityChanged(EventHandler`1 value) in D:\a\_work\1\s\src\Essentials\src\Connectivity\Connectivity.shared.cs:91
    at Microsoft.Maui.Networking.Connectivity.add_ConnectivityChanged(EventHandler`1 value) in D:\a\_work\1\s\src\Essentials\src\Connectivity\Connectivity.shared.cs:58
    at DotnetMaui.App..ctor() in /Users/daniel/Development/Samples/DotnetMaui/App.xaml.cs:9
```

### Steps to Reproduce

1. Create a new MAUI app with botnet new maui
2. In the App constructor in App.xaml.cs, add :
Connectivity.ConnectivityChanged += (sender, e) => {   };
3. Start the app on an Android 14 emulator or device.


### Link to public reproduction project repository

_No response_

### Version with bug

8.0.0-rc.1.9171

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 14

### Did you find any workaround?

Change the target Android version to 13. The code will work on Android 14 then.

### Relevant log output

```shell
Exception has occurred: Java.Lang.SecurityException
com.companyname.dotnetmaui: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts
    at Java.Interop.JniEnvironment.InstanceMethods.CallNonvirtualObjectMethod(JniObjectReference instance, JniObjectReference type, JniMethodInfo method, JniArgumentValue* args) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/obj/Release/net7.0/JniEnvironment.g.cs:20416
    at Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeVirtualObjectMethod(String encodedMember, IJavaPeerable self, JniArgumentValue* parameters) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.JniInstanceMethods_Invoke.cs:876
    at Android.Content.ContextWrapper.RegisterReceiver(BroadcastReceiver receiver, IntentFilter filter) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Android.Content.ContextWrapper.cs:2605
    at Microsoft.Maui.Networking.ConnectivityImplementation.StartListeners() in D:\a\_work\1\s\src\Essentials\src\Connectivity\Connectivity.android.cs:48
    at Microsoft.Maui.Networking.ConnectivityImplementation.add_ConnectivityChanged(EventHandler`1 value) in D:\a\_work\1\s\src\Essentials\src\Connectivity\Connectivity.shared.cs:91
    at Microsoft.Maui.Networking.Connectivity.add_ConnectivityChanged(EventHandler`1 value) in D:\a\_work\1\s\src\Essentials\src\Connectivity\Connectivity.shared.cs:58
    at DotnetMaui.App..ctor() in /Users/daniel/Development/Samples/DotnetMaui/App.xaml.cs:9
    at at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Constructor(Object obj, IntPtr* args)
    at at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
    at at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
    at at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
    at at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSiteMain(ServiceCallSite callSite, RuntimeResolverContext argument)
    at at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
    at at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSite(ServiceCallSite callSite, RuntimeResolverContext argument)
    at at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
    at at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(ServiceIdentifier serviceIdentifier)
    at at System.Collections.Concurrent.ConcurrentDictionary`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceIdentifier, Microsoft.Extensions.DependencyInjection, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[Microsoft.Extensions.DependencyInjection.ServiceProvider.ServiceAccessor, Microsoft.Extensions.DependencyInjection, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].GetOrAdd(ServiceIdentifier key, Func`2 valueFactory)
    at at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(ServiceIdentifier serviceIdentifier, ServiceProviderEngineScope serviceProviderEngineScope)
    at at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType)
    at Microsoft.Maui.MauiContext.WrappedServiceProvider.GetService(Type serviceType) in D:\a\_work\1\s\src\Core\src\MauiContext.cs:68
    at Microsoft.Maui.MauiContext.WrappedServiceProvider.GetService(Type serviceType) in D:\a\_work\1\s\src\Core\src\MauiContext.cs:68
    at at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
    at at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[IApplication](IServiceProvider provider)
    at Microsoft.Maui.MauiApplication.OnCreate() in D:\a\_work\1\s\src\Core\src\Platform\Android\MauiApplication.cs:46
    at Android.App.Application.n_OnCreate(IntPtr jnienv, IntPtr native__this) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net8.0/android-34/mcw/Android.App.Application.cs:1086
    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V(_JniMarshal_PP_V callback, IntPtr jnienv, IntPtr klazz) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:26
    at java.lang.SecurityException: com.companyname.dotnetmaui: One of RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED should be specified when a receiver isn't being registered exclusively for system broadcasts
    at at android.os.Parcel.createExceptionOrNull(Parcel.java:3057)
    at at android.os.Parcel.createException(Parcel.java:3041)
    at at android.os.Parcel.readException(Parcel.java:3024)
    at at android.os.Parcel.readException(Parcel.java:2966)
    at at android.app.IActivityManager$Stub$Proxy.registerReceiverWithFeature(IActivityManager.java:5684)
    at at android.app.ContextImpl.registerReceiverInternal(ContextImpl.java:1852)
    at at android.app.ContextImpl.registerReceiver(ContextImpl.java:1792)
    at at android.app.ContextImpl.registerReceiver(ContextImpl.java:1780)
    at at android.content.ContextWrapper.registerReceiver(ContextWrapper.java:755)
    at at crc6488302ad6e9e4df1a.MauiApplication.n_onCreate(Native Method)
    at at crc6488302ad6e9e4df1a.MauiApplication.onCreate(MauiApplication.java:28)
    at at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1316)
    at at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6998)
    at at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
    at at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2236)
    at at android.os.Handler.dispatchMessage(Handler.java:106)
    at at android.os.Looper.loopOnce(Looper.java:205)
    at at android.os.Looper.loop(Looper.java:294)
    at at android.app.ActivityThread.main(ActivityThread.java:8177)
    at at java.lang.reflect.Method.invoke(Native Method)
    at at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
    at at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
    at Caused by: android.os.RemoteException: Remote stack trace:
    at at com.android.server.am.ActivityManagerService.registerReceiverWithFeature(ActivityManagerService.java:13908)
    at at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:2570)
    at at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2720)
    at at android.os.Binder.execTransactInternal(Binder.java:1339)
    at at android.os.Binder.execTransact(Binder.java:1275)
```

--- a/src/Essentials/src/Battery/Battery.android.cs
+++ b/src/Essentials/src/Battery/Battery.android.cs
@@ -19,7 +19,7 @@ namespace Microsoft.Maui.Devices
 		void StartEnergySaverListeners()
 		{
 			powerReceiver = new EnergySaverBroadcastReceiver(OnEnergySaverChanged);
-			Application.Context.RegisterReceiver(powerReceiver, new IntentFilter(PowerManager.ActionPowerSaveModeChanged));
+			PlatformUtils.RegisterBroadcastReceiver(powerReceiver, new IntentFilter(PowerManager.ActionPowerSaveModeChanged), false);
 		}
 
 		void StopEnergySaverListeners()
@@ -50,7 +50,7 @@ namespace Microsoft.Maui.Devices
 			Permissions.EnsureDeclared<Permissions.Battery>();
 
 			batteryReceiver = new BatteryBroadcastReceiver(OnBatteryInfoChanged);
-			Application.Context.RegisterReceiver(batteryReceiver, new IntentFilter(Intent.ActionBatteryChanged));
+			PlatformUtils.RegisterBroadcastReceiver(batteryReceiver, new IntentFilter(Intent.ActionBatteryChanged), false);
 		}
 
 		void StopBatteryListeners()
@@ -74,7 +74,7 @@ namespace Microsoft.Maui.Devices
 				Permissions.EnsureDeclared<Permissions.Battery>();
 
 				using (var filter = new IntentFilter(Intent.ActionBatteryChanged))
-				using (var battery = Application.Context.RegisterReceiver(null, filter))
+				using (var battery = PlatformUtils.RegisterBroadcastReceiver(null, filter, false))
 				{
 					var level = battery.GetIntExtra(BatteryManager.ExtraLevel, -1);
 					var scale = battery.GetIntExtra(BatteryManager.ExtraScale, -1);
@@ -94,7 +94,7 @@ namespace Microsoft.Maui.Devices
 				Permissions.EnsureDeclared<Permissions.Battery>();
 
 				using (var filter = new IntentFilter(Intent.ActionBatteryChanged))
-				using (var battery = Application.Context.RegisterReceiver(null, filter))
+				using (var battery = PlatformUtils.RegisterBroadcastReceiver(null, filter, false))
 				{
 					var status = battery.GetIntExtra(BatteryManager.ExtraStatus, -1);
 					switch (status)
@@ -121,7 +121,7 @@ namespace Microsoft.Maui.Devices
 				Permissions.EnsureDeclared<Permissions.Battery>();
 
 				using (var filter = new IntentFilter(Intent.ActionBatteryChanged))
-				using (var battery = Application.Context.RegisterReceiver(null, filter))
+				using (var battery = PlatformUtils.RegisterBroadcastReceiver(null, filter, false))
 				{
 					var chargePlug = battery.GetIntExtra(BatteryManager.ExtraPlugged, -1);
 

--- a/src/Essentials/src/Connectivity/Connectivity.android.cs
+++ b/src/Essentials/src/Connectivity/Connectivity.android.cs
@@ -45,7 +45,7 @@ namespace Microsoft.Maui.Networking
 
 			conectivityReceiver = new ConnectivityBroadcastReceiver(OnConnectivityChanged);
 
-			Application.Context.RegisterReceiver(conectivityReceiver, filter);
+			PlatformUtils.RegisterBroadcastReceiver(conectivityReceiver, filter, true);
 		}
 
 		void StopListeners()

--- a/src/Essentials/src/Connectivity/Connectivity.android.cs
+++ b/src/Essentials/src/Connectivity/Connectivity.android.cs
@@ -45,7 +45,7 @@ namespace Microsoft.Maui.Networking
 
 			conectivityReceiver = new ConnectivityBroadcastReceiver(OnConnectivityChanged);
 
-			PlatformUtils.RegisterBroadcastReceiver(conectivityReceiver, filter, true);
+			PlatformUtils.RegisterBroadcastReceiver(conectivityReceiver, filter, false);
 		}
 
 		void StopListeners()

--- a/src/Essentials/src/Platform/PlatformUtils.android.cs
+++ b/src/Essentials/src/Platform/PlatformUtils.android.cs
@@ -1,3 +1,4 @@
+#nullable enable
 using System;
 using Android.App;
 using Android.Content;
@@ -28,10 +29,13 @@ namespace Microsoft.Maui.ApplicationModel
 		internal static bool HasSystemFeature(string systemFeature)
 		{
 			var packageManager = Application.Context.PackageManager;
-			foreach (var feature in packageManager.GetSystemAvailableFeatures())
+			if (packageManager is not null)
 			{
-				if (feature?.Name?.Equals(systemFeature, StringComparison.OrdinalIgnoreCase) ?? false)
-					return true;
+				foreach (var feature in packageManager.GetSystemAvailableFeatures())
+				{
+					if (feature?.Name?.Equals(systemFeature, StringComparison.OrdinalIgnoreCase) ?? false)
+						return true;
+				}
 			}
 			return false;
 		}
@@ -52,37 +56,16 @@ namespace Microsoft.Maui.ApplicationModel
 			return intent.ResolveActivity(pm) is ComponentName c && c.PackageName == expectedPackageName;
 		}
 
-		internal static Java.Util.Locale GetLocale()
+		internal static Intent? RegisterBroadcastReceiver(BroadcastReceiver? receiver, IntentFilter filter, bool exported)
 		{
-			var resources = Application.Context.Resources;
-			var config = resources.Configuration;
-
-#if __ANDROID_24__
-			if (OperatingSystem.IsAndroidVersionAtLeast(24))
-				return config.Locales.Get(0);
+#if ANDROID34_0_OR_GREATER
+			if (OperatingSystem.IsAndroidVersionAtLeast(34))
+			{
+				var flags = exported ? ReceiverFlags.Exported : ReceiverFlags.NotExported;
+				return Application.Context.RegisterReceiver(receiver, filter, flags);
+			}
 #endif
-
-#pragma warning disable CS0618 // Type or member is obsolete
-			return config.Locale;
-#pragma warning restore CS0618 // Type or member is obsolete
-		}
-
-		internal static void SetLocale(Java.Util.Locale locale)
-		{
-			Java.Util.Locale.Default = locale;
-			var resources = Application.Context.Resources;
-			var config = resources.Configuration;
-#pragma warning disable CS0618 // Type or member is obsolete
-#pragma warning disable CA1422 // Validate platform compatibility
-#pragma warning disable CA1416 // Validate platform compatibility
-			if (OperatingSystem.IsAndroidVersionAtLeast(24))
-				config.SetLocale(locale);
-			else
-				config.Locale = locale;
-			resources.UpdateConfiguration(config, resources.DisplayMetrics);
-#pragma warning restore CA1422 // Validate platform compatibility
-#pragma warning restore CA1416 // Validate platform compatibility
-#pragma warning restore CS0618 // Type or member is obsolete
+			return Application.Context.RegisterReceiver(receiver, filter);
 		}
 	}
 }

--- a/src/Essentials/src/Platform/PlatformUtils.android.cs
+++ b/src/Essentials/src/Platform/PlatformUtils.android.cs
@@ -26,6 +26,18 @@ namespace Microsoft.Maui.ApplicationModel
 			return requestCode;
 		}
 
+		internal static Intent? RegisterBroadcastReceiver(BroadcastReceiver? receiver, IntentFilter filter, bool exported)
+		{
+#if ANDROID34_0_OR_GREATER
+			if (OperatingSystem.IsAndroidVersionAtLeast(34))
+			{
+				var flags = exported ? ReceiverFlags.Exported : ReceiverFlags.NotExported;
+				return Application.Context.RegisterReceiver(receiver, filter, flags);
+			}
+#endif
+			return Application.Context.RegisterReceiver(receiver, filter);
+		}
+
 		internal static bool HasSystemFeature(string systemFeature)
 		{
 			var packageManager = Application.Context.PackageManager;
@@ -54,18 +66,6 @@ namespace Microsoft.Maui.ApplicationModel
 				return false;
 
 			return intent.ResolveActivity(pm) is ComponentName c && c.PackageName == expectedPackageName;
-		}
-
-		internal static Intent? RegisterBroadcastReceiver(BroadcastReceiver? receiver, IntentFilter filter, bool exported)
-		{
-#if ANDROID34_0_OR_GREATER
-			if (OperatingSystem.IsAndroidVersionAtLeast(34))
-			{
-				var flags = exported ? ReceiverFlags.Exported : ReceiverFlags.NotExported;
-				return Application.Context.RegisterReceiver(receiver, filter, flags);
-			}
-#endif
-			return Application.Context.RegisterReceiver(receiver, filter);
 		}
 	}
 }

--- a/src/Essentials/test/DeviceTests/Tests/Battery_Tests.cs
+++ b/src/Essentials/test/DeviceTests/Tests/Battery_Tests.cs
@@ -1,4 +1,5 @@
 using Microsoft.Maui.Devices;
+using System.Threading.Tasks;
 using Xunit;
 
 namespace Microsoft.Maui.Essentials.DeviceTests
@@ -52,6 +53,38 @@ namespace Microsoft.Maui.Essentials.DeviceTests
 		public void App_Is_Not_Lower_Power_mode()
 		{
 			Assert.Equal(EnergySaverStatus.Off, Battery.EnergySaverStatus);
+		}
+
+		[Fact]
+		public async Task EnergySaverStatusChanged_Does_Not_Crash()
+		{
+			Battery.EnergySaverStatusChanged += Battery_EnergySaverStatusChanged;
+
+			// just ensure there is no need for the OS to "respond" to a new subscription
+			await Task.Delay(1000);
+
+			Battery.EnergySaverStatusChanged -= Battery_EnergySaverStatusChanged;
+
+			static void Battery_EnergySaverStatusChanged(object sender, EnergySaverStatusChangedEventArgs e)
+			{
+				// do nothing
+			}
+		}
+
+		[Fact]
+		public async Task BatteryInfoChanged_Does_Not_Crash()
+		{
+			Battery.BatteryInfoChanged += Battery_BatteryInfoChanged;
+
+			// just ensure there is no need for the OS to "respond" to a new subscription
+			await Task.Delay(1000);
+
+			Battery.BatteryInfoChanged -= Battery_BatteryInfoChanged;
+
+			static void Battery_BatteryInfoChanged(object sender, BatteryInfoChangedEventArgs e)
+			{
+				// do nothing
+			}
 		}
 	}
 }

--- a/src/Essentials/test/DeviceTests/Tests/Connectivity_Tests.cs
+++ b/src/Essentials/test/DeviceTests/Tests/Connectivity_Tests.cs
@@ -24,6 +24,22 @@ namespace Microsoft.Maui.Essentials.DeviceTests
 		}
 
 		[Fact]
+		public async Task ConnectivityChanged_Does_Not_Crash()
+		{
+			Connectivity.ConnectivityChanged += Connectivity_ConnectivityChanged;
+
+			// just ensure there is no need for the OS to "respond" to a new subscription
+			await Task.Delay(1000);
+
+			Connectivity.ConnectivityChanged -= Connectivity_ConnectivityChanged;
+
+			static void Connectivity_ConnectivityChanged(object sender, ConnectivityChangedEventArgs e)
+			{
+				// do nothing
+			}
+		}
+
+		[Fact]
 		public async Task Test()
 		{
 			var current = Connectivity.Current.NetworkAccess;

